### PR TITLE
Fix interactions in text-in-voice channels

### PR DESCRIPTION
**Pull request by Mysterypotatoguy**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](https://github.com/Javacord/Javacord/blob/master/.github/CONTRIBUTING.md).

## Changelog
- Fixed AssertionError when receiving interactions from text-in-voice channels

Closes: #1222 

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

File: javacord-core/src/main/java/org/javacord/core/util/handler/interaction/InteractionCreateHandler.java [hunk 1]
```diff
@@ -64,7 +65,9 @@ public void handle(JsonNode packet) {
 
             // Check if this interaction comes from a guild or a DM
             if (packet.hasNonNull("guild_id")) {
-                channel = api.getTextChannelById(channelId).orElse(null);
+                channel = api.getServerChannelById(channelId)
+                        .flatMap(ServerChannel::asTextableRegularServerChannel)
+                        .orElse(null);
```
Comment (felldo) [inline javacord-core/src/main/java/org/javacord/core/util/handler/interaction/InteractionCreateHandler.java:70]: Makes it easier to debug next time
```suggestion
                        .orElseThrow(AssertionError::new);
```

**Review by felldo — CHANGES_REQUESTED** [review]

**Review by felldo — COMMENTED** [review]
Not exactly sure what I am doing wrong, but if I send a slash command in the voice channel as described in the issue and I do not get any error. And since every TextableRegularServerChannel is a TextChannel, we should be able to get the voice channel

**Comment by Mysterypotatoguy** [comment]
Well that's super odd, I was able to reproduce it earlier and now I cannot 🤔
Perhaps this is some caching issue with text-in-voice channels instead?

**Comment by felldo** [comment]
> Well that's super odd, I was able to reproduce it earlier and now I cannot 🤔 Perhaps this is some caching issue with text-in-voice channels instead?

This would be my first guess, that the channel is not even cached

**Comment by RealMarvn** [comment]
Ok so I have other commands which try to get the voice channel by ID. Those work great most of the time. The only leak is when I try to get the voice channel via getConnectedVoiceChannel. It could be a cache issue. But I have never watched inside Javacord.

**Pull request closed without merging**